### PR TITLE
DM-27177: Remove lsst.afw.image.Filter

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR removes all `Filter` support that had been kept for the transition period. It replaces the old `filter` component with a new component (backed by `afw.image.FilterLabel`), and deprecates the transitional `filterLabel` component.

### CODE REVIEW

File: python/lsst/obs/base/_fitsRawFormatterBase.py [hunk 1]
```diff
@@ -323,8 +298,11 @@ def readComponent(self, component):
         if component == "image":
             return self.readImage()
         elif component == "filter":
-            return self.makeFilter()
+            return self.makeFilterLabel()
+        # TODO: remove in DM-27811
         elif component == "filterLabel":
+            warnings.warn("Exposure.filterLabel component is deprecated; use .filter instead. "
+                          "Will be removed after v25.")
```
Comment (timj) [inline python/lsst/obs/base/_fitsRawFormatterBase.py:305]: I think this has to be a FutureWarning.

**Comment by timj** [comment]
@kfindeisen the advice is to let it make the change. Do not fight black.

**Comment by kfindeisen** [comment]
I'm not trying to "fight" anything. I'm asking how to deal with things like:

```-    EXPOSURE_READ_COMPONENTS = {"bbox", "dimensions", "xy0", "filterLabel", }
+    EXPOSURE_READ_COMPONENTS = {
+        "bbox",
+        "dimensions",
+        "xy0",
+        "filterLabel",
+    }
```

Should I make a change like this when I first touch the original line, when I last touch it, in a separate commit before any of the substantial ones, etc.?

**Comment by timj** [comment]
@kfindeisen at this point do a new commit. The advice in middleware packages is to use `pre-commit` so that you don't have to worry about it again because it will run black for you and ensure that every commit is compliant. 

https://developer.lsst.io/python/formatting.html#using-pre-commit

**Review by TallJimbo — APPROVED** [review]

**Pull request merged**